### PR TITLE
Fix wrong accumulation with average='none'

### LINKED ISSUE #1045: Accuracy, Precision, Recall, Specificity wrong accumulation with average='none' inside Wrapper.
## 🐛 Bug

The Classification Metrics Accuracy, Precision, Recall and Specificity in average='none' mode return negative metrics values, when called multiple times once with missing and once with present targets. 

### To Reproduce

```py
import torchmetrics as tm
import torch


class MetricWrapper(tm.Metric):
    def __init__(self, metric):
        super().__init__()
        self.metric = metric
    
    def update(self, *args, **kwargs):
        self.metric.update(*args, **kwargs)
    
    def compute(self, *args, **kwargs):
        return self.metric.compute(*args, **kwargs)



def main():
    acc = MetricWrapper(tm.Accuracy(num_classes=3, average='none'))
    # EDIT!
    # pred = torch.tensor([[0., 1., 0.], [0., 0., 1.]])
    # to
    pred = torch.tensor([[0., 1., 0.], [1., 0., 0.]])
    target = torch.tensor([0, 1])

    pred2 = torch.tensor([[0., 1., 0.], [1., 0., 0.]])
    target2 = torch.tensor([0, 2])

    print(acc(pred, target))

    print(acc(pred2, target2))


if __name__ == "__main__":
    main()
```

Output:
tensor([0., 0., nan])
tensor([0., 0., -1.])

### Expected behavior

Output:
tensor([0., 0., nan])
tensor([0., 0., 0.])

### Environment

- TorchMetrics version: 0.8.2

### Additional context
The origin of the problem is already found: Bad handling of mutability.
PR follows.

### PR DESCRIPTION
**Pull request by kpoeppel**
## What does this PR do?

Fixes #1045, simply copying tensors before they are mutated in-place.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**? 
 -> Feel free to add the test mentioned in #1045

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — CHANGES_REQUESTED** [review]
pls add the test mentioned in #1045

**Comment by kpoeppel** [comment]
@SkafteNicki :
I am sorry, somehow I added a wrong bug report code, it is updated now and I also checked it on master now.

@Borda: 
Test will be added now. For the specificity there is no error currently, but it is bad structure to assume that tn can be mutated.

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]
